http/tcp checks: fix long timeout behavior to default to user-configured value (#6094)
hashicorp/consul: website/source/docs/agent/checks.html.md

  blog post](https://www.hashicorp.com/blog/protecting-consul-from-rce-risk-in-specific-configurations)
  for more details.

* HTTP + Interval - These checks make an HTTP `GET` request to the specified URL,

>   waiting the specified Interval amount of time (eg. 30 seconds) between requests.

Reviewer: I think this reads a little easier if the `eg` isn't in the middle of the sentence:
```suggestion
  waiting between requests for the specified interval time (eg. 30 seconds).
```

Also we probably shouldn't be capitalizing "interval" in the description anymore.

Author: sounds good! shall I codify interval like `timeout` is? (`interval`)? that's sort of what i was expecting in the docs

  The status of the service depends on the HTTP response code: any `2xx` code is 
  considered passing, a `429 Too ManyRequests` is a warning, and anything else is
  a failure. This type of check
  should be preferred over a script that uses `curl` or another external process
  to check a simple HTTP operation. By default, HTTP checks are `GET` requests
  unless the `method` field specifies a different method. Additional header
  fields can be set through the `header` field which is a map of lists of
  strings, e.g. `{"x-foo": ["bar", "baz"]}`. By default, HTTP checks will be
  configured with a request timeout equal to 10 seconds.
  It is possible to configure a custom HTTP check timeout value by
  specifying the `timeout` field in the check definition. The output of the
  check is limited to roughly 4KB. Responses larger than this will be truncated.
  HTTP checks also support TLS. By default, a valid TLS certificate is expected.
  Certificate verification can be turned off by setting the `tls_skip_verify`
  field to `true` in the check definition.

> * TCP + Interval - These checks make a TCP connection attempt to the specified 

Reviewer: Same as above.

* gRPC + Interval - These checks are intended for applications that support the standard
  [gRPC health checking protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md).
  The state of the check will be updated by probing the configured endpoint, waiting Interval

>   amount of time (eg. 30 seconds) between probes. By default, gRPC checks will be configured 

Reviewer: Same as above.
